Example/Tests/Tests.m

```diff
@@ -412,28 +412,33 @@ - (ADBMediaObject *)createWithProperties:(NSDictionary *)properties
+        __block ADBMediaHeartbeat *mockADBMediaHeartbeat;
+        ADBMediaHeartbeatConfig *config = [[ADBMediaHeartbeatConfig alloc] init];
+        __block ADBMediaObject *mockADBMediaObject;
 
-            beforeEach(^{
-                mockADBMediaHeartbeat = mock([ADBMediaHeartbeat class]);
-                mockADBMediaObject = mock([ADBMediaObject class]);
+        beforeEach(^{
+            mockADBMediaHeartbeat = mock([ADBMediaHeartbeat class]);
+            mockADBMediaObject = mock([ADBMediaObject class]);
 
-                SEGMockADBMediaHeartbeatFactory *mockADBMediaHeartbeatFactory = [[SEGMockADBMediaHeartbeatFactory alloc] init];
-                mockADBMediaHeartbeatFactory.ADBMediaHeartbeat = mockADBMediaHeartbeat;
+            SEGMockADBMediaHeartbeatFactory *mockADBMediaHeartbeatFactory = [[SEGMockADBMediaHeartbeatFactory alloc] init];
+            mockADBMediaHeartbeatFactory.ADBMediaHeartbeat = mockADBMediaHeartbeat;
 
-                SEGMockADBMediaObjectFactory *mockADBMediaObjectFactory = [[SEGMockADBMediaObjectFactory alloc] init];
-                mockADBMediaObjectFactory.ADBMediaObject = mockADBMediaObject;
+            SEGMockADBMediaObjectFactory *mockADBMediaObjectFactory = [[SEGMockADBMediaObjectFactory alloc] init];
+            mockADBMediaObjectFactory.ADBMediaObject = mockADBMediaObject;
 
-                integration = [[SEGAdobeIntegration alloc] initWithSettings:@{ @"heartbeatTrackingServer" : @"example",
-                                                                               @"ssl" : @YES }
-                                                               andADBMobile:mockADBMobile
-                                                andADBMediaHeartbeatFactory:mockADBMediaHeartbeatFactory
-                                                 andADBMediaHeartbeatConfig:config
-                                                   andADBMediaObjectFactory:mockADBMediaObjectFactory];
-            });
+            integration = [[SEGAdobeIntegration alloc] initWithSettings:@{ @"heartbeatTrackingServer" : @"example",
+                                                                           @"ssl" : @YES }
+                                                           andADBMobile:mockADBMobile
+                                            andADBMediaHeartbeatFactory:mockADBMediaHeartbeatFactory
+                                             andADBMediaHeartbeatConfig:config
+                                               andADBMediaObjectFactory:mockADBMediaObjectFactory];
+
+            // Video Playback Started initializes an instance of ADBMediaHeartbeat, which we need for testing subsequent Video Events
+            SEGTrackPayload *payload = [[SEGTrackPayload alloc] initWithEvent:@"Video Playback Started" properties:@{} context:@{}
+                integrations:@{}];
+            [integration track:payload];
+        });
+        describe(@"initialization", ^{
```

Comment:
Not a blocker, but initialization to me suggests testing the integration inititalization, not that specific events do some initialization. I think the event initialization should be under a different sub test.

Reply:
Not sure I understand, should I just rename this? It can be initialization on Video Playback Started